util/env_posix.cc

```diff
@@ -261,6 +267,9 @@ class PosixEnv : public Env {
       // appends data to the end of the file, regardless of the value of
       // offset.
       // More info here: https://linux.die.net/man/2/pwrite
+#ifdef ROCKSDB_LITE
+      return Status::IOError(fname, "Direct I/O not supported in RocksDB lite");
+#endif  // !ROCKSDB_LITE
```

Comment:
should there be an #else here? i thought the compiler would warn if there's code underneath a return since it's unreachable.